backend-config/namespace_config.go

```diff
@@ -168,7 +168,9 @@ func (nc *namespaceConfig) getFromAPI(ctx context.Context) (map[string]ConfigT,
 		} else {
 			workspace.ApplyReplaySources()
 		}
-
+		if workspace != nil {
+			workspace.processAccountAssociations()
+		}
```

Comment:
simply move `workspace.processAccountAssociations()` inside the `else` branch above
```suggestion
```